fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java

```diff
@@ -236,10 +236,18 @@ public InternalCatalog() {
         mysqlCompatibleDatabases.add(new MysqlDb());
         MysqlCompatibleDatabase.COUNT = 2;
 
-        for (MysqlCompatibleDatabase idb : mysqlCompatibleDatabases) {
-            // do not call unprotectedCreateDb, because it will cause loop recursive when initializing Env singleton
-            idToDb.put(idb.getId(), idb);
-            fullNameToDb.put(idb.getFullName(), idb);
+        if (!tryLock(false)) {
```

Comment:
in constructor  no need lock because at this time, no other thread can access this object yet